crates/nix_rs/src/flake/metadata.rs

```diff
@@ -0,0 +1,19 @@
+use crate::command::{NixCmd, NixCmdError};
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct FlakeMetadata {
+    pub path: String,
+}
+
+/// Runs `nix flake metadata json` in Rust
+pub async fn get_flake_metadata_json(
```

Comment:
This should following the naming convention (see `config.rs`, `outputs.rs` for example).